doc/md: guide for sequelize schema loader (#2098)
ariga/atlas: doc/md/guides/orms/sequelize.mdx

* Developers using Sequelize can use Atlas to automatically plan schema migrations
  for them, based on the desired state of their schema instead of crafting them by hand.

## Automatic migration planning for Sequelize
Sequelize is a popular ORM widely used in the Node.js community. Sequelize allows users to
manage their database schemas using its [sync](https://sequelize.org/docs/v6/core-concepts/model-basics/#model-synchronization/) feature,
which is usually sufficient during development and in many simple cases.

However, at some point, teams need more control and decide to employ
the [migrations](https://sequelize.org/master/manual/migrations.html)
methodology, which is a more robust way to manage your database schema.
The problem with [creating migrations](https://sequelize.org/docs/v6/other-topics/migrations/#creating-the-first-model-and-migration)
in Sequelize is that they are usually written by hand, which is error-prone and
time-consuming.

Atlas can automatically plan database schema migrations for developers using Sequelize.
Atlas plans migrations by calculating the diff between the _current_ state of the database,
and its _desired_ state.

> In the context of [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations), the current state can be thought of as the database schema that would have
> been created by applying all previous migration scripts.

Reviewer: pay attention to line length
```suggestion
In the context of [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations),
the current state can be thought of as the database schema that would have been created by applying
all previous migration scripts.
```

The desired schema of your application can be provided to Atlas via an [External Schema Datasource](/atlas-schema/projects#data-source-external_schema)
which is any program that can output a SQL schema definition to stdout.

To use Atlas with Sequelize, users can utilize the [Sequelize Atlas Provider](https://github.com/ariga/atlas-provider-sequelize),
a small program that can be used to load the schema of a Sequelize project into Atlas.

In this guide, we will show how Atlas can be used to automatically plan schema migrations for
Sequelize users.

## Prerequisites

> * A local [sequelize](https://github.com/sequelize/sequelize) project.

Reviewer: ```suggestion
* A local [Sequelize](https://github.com/sequelize/sequelize) project.
```

If you don't have a Sequelize project handy, you can use [sequelize/express-example](https://github.com/sequelize/express-example)
as a starting point:

```
git clone git@github.com:sequelize/express-example.git
```

## Using the Atlas Sequelize Provider

> In this guide, we will use the [Sequelize Atlas Provider](https://github.com/ariga/atlas-provider-sequelize)

Reviewer: ```suggestion

In this guide, we will use the [Sequelize Atlas Provider](https://github.com/ariga/atlas-provider-sequelize)
```

to automatically plan schema migrations for a Sequelize project.

### Installation

Install Atlas from macOS or Linux by running:
```bash
curl -sSf https://atlasgo.sh | sh
```
See [atlasgo.io](https://atlasgo.io/getting-started#installation) for more installation options.

Install the provider by running:

<Tabs>
<TabItem value="js" label="JavaScript" default>

```bash
npm install @ariga/atlas-provider-sequelize
```
</TabItem>
<TabItem value="ts" label="TypeScript">

```bash
npm install @ariga/ts-atlas-provider-sequelize
```
</TabItem>
</Tabs>

> Make sure all you Node dependencies are installed by running:

Reviewer: ```suggestion
Make sure all your Node dependencies are installed by running:
```

```bash
npm install
```

### Standalone vs Script mode

The Atlas Sequelize Provider can be used in two modes:

> * **Standalone** - If all of your Sequelize models exist in single directory,
>  you can use the provider directly to load your Sequelize schema into Atlas.

Reviewer: I think a directory is called a module in this context? 
```suggestion
* **Standalone** - If all of your Sequelize models exist in a single Node module,
 you can use the provider directly to load your Sequelize schema into Atlas.
```

> * **Script** - In other cases, you can use the provider as a npm package to write a script
>  that loads your Sequelize schema into Atlas.

Reviewer: ```suggestion
* **Script** - In other cases, you can use the provider as an npm package to write a script
 that loads your Sequelize schema into Atlas.
```

```
</TabItem>
</Tabs>
</TabItem>
</Tabs>

Using the [Standalone mode](#standalone-mode) configuration file for the Atlas Sequelize Provider,
we can generate a migration file by running this command:

```bash
atlas migrate diff --env sequelize
```

Running this command will generate files similar to this in the `migrations` directory:

```
migrations
|-- 20230918143104.sql
`-- atlas.sum

0 directories, 2 files
```

Examining the contents of `20230918143104.sql`:

```sql
-- Create "Users" table
CREATE TABLE `Users` (
  `id` int NOT NULL AUTO_INCREMENT,
  `name` varchar(255) NOT NULL,
  `email` varchar(255) NOT NULL,
  `createdAt` datetime NOT NULL,
  `updatedAt` datetime NOT NULL,
  PRIMARY KEY (`id`)
) CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci;
-- Create "Tasks" table
CREATE TABLE `Tasks` (
  `id` int NOT NULL AUTO_INCREMENT,
  `complete` bool NULL DEFAULT 0,
  `createdAt` datetime NOT NULL,
  `updatedAt` datetime NOT NULL,
  `userID` int NOT NULL,
  PRIMARY KEY (`id`),
  INDEX `userID` (`userID`),
  CONSTRAINT `Tasks_ibfk_1` FOREIGN KEY (`userID`) REFERENCES `Users` (`id`) ON UPDATE CASCADE ON DELETE NO ACTION
) CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci;
```

> Amazing, Atlas automatically generated a migration file that will create the `Users` and `Tasks` tables in our database!

Reviewer: ```suggestion
Amazing! Atlas automatically generated a migration file that will create the `Users` and `Tasks` tables in our database!
```
